src/components/Input/Input.tsx

```diff
@@ -104,120 +110,123 @@ class Input extends AutoControlledComponent<Extendable<IInputProps>, any> {
      */
     renderInput: PropTypes.func,
 
+    /**
+     * A custom render function the wrapper slot.
+     *
+     * @param { React.ReactType } Component - The computed component for this slot.
+     * @param { object } props - The computed props for this slot.
+     * @param { ReactNode | ReactNodeArray } children - The computed children for this slot.
+     */
+    renderWrapper: PropTypes.func,
+
     /** Additional CSS styles to apply to the component instance.  */
     styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
 
+    /** The HTML input type. */
+    type: PropTypes.string,
+
     /** The value of the input. */
     value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
 
     /** Override for theme site variables to allow modifications of component styling via themes. */
     variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Shorthand for the wrapper component */
+    wrapper: customPropTypes.itemShorthand,
   }
 
   static defaultProps = {
     as: 'div',
     type: 'text',
+    wrapper: 'div',
   }
 
   static autoControlledProps = ['value']
 
-  inputRef: any
+  state = { value: this.props.value || this.props.defaultValue || '' }
+
+  renderComponent({
+    ElementType,
+    classes,
+    rest: restProps,
+    styles,
+    variables,
+  }: IRenderResultConfig<IInputProps>) {
+    const { className, input, renderIcon, renderInput, renderWrapper, type, wrapper } = this.props
+    const { value = '' } = this.state
```

Comment:
Do we need default value - aren't we doing already in 

state = { value: this.props.value || this.props.defaultValue || '' }